app/models/manageiq/providers/vmware/cloud_manager/orchestration_service_option_converter.rb

```diff
@@ -3,77 +3,122 @@ class Vmware::CloudManager::OrchestrationServiceOptionConverter < ::ServiceOrche
     include Vmdb::Logging
 
     def stack_create_options
+      @template = ManageIQ::Providers::Vmware::CloudManager::OvfTemplate.new(self.class.get_template(@dialog_options).content)
       options = {
         :deploy  => stack_parameters['deploy'] == 't',
         :powerOn => stack_parameters['powerOn'] == 't'
       }
       options[:vdc_id] = @dialog_options['dialog_availability_zone'] unless @dialog_options['dialog_availability_zone'].blank?
-
-      options.merge!(customize_vapp_template(collect_vm_params))
+      options.merge!(customize_vapp_template(collect_vm_params, collect_vapp_net_params))
     end
 
     private
 
     # customize_vapp_template will prepare the options in a format suitable for the fog-vcloud-director.
-    # This mainly results in creating two top level objects in a hash. The :InstantiationParams, contains
-    # a single :NetworkConfig element with the array of all references to the networks used in the
-    # deployed vApp. We are using IDs here and let fog library create concrete HREFs that are required
-    # by the vCloud director. The second object is an array of source items. Each source item references
-    # a single VM from the vApp template, customises its name and optionally sets network info.
-    def customize_vapp_template(vm_params)
-      network_config = {}
-
-      source_vms = vm_params.collect do |vm_id, vm_opts|
-        src_vm = { :vm_id => "vm-#{vm_id}" }
-        src_vm[:name] = vm_opts["instance_name"] if vm_opts.key?("instance_name")
-
-        network_id = vm_opts["vdc_network"]
-        unless network_id.nil?
-          # Create new network config if it hasn't been created before.
-          network_config[network_id] ||= {
-            :networkName => network_id,
-            :networkId   => network_id,
-            :fenceMode   => "bridged"
+    # See https://github.com/xlab-si/fog-vcloud-director/blob/master/docs/examples-vapp-instantiate.md
+    def customize_vapp_template(vm_params, vapp_net_params)
+      source_vms = vm_params.map do |_, vm_opts|
+        src_vm = {
+          :vm_id    => "vm-#{vm_opts[:vm_id]}",
+          :networks => parse_nics(vm_opts),
+          :hardware => {
+            :cpu    => { :num_cores => vm_opts['num_cores'], :cores_per_socket => vm_opts['cores_per_socket'] },
+            :memory => { :quantity_mb => vm_opts['memory_mb'] }
           }
-
-          # Add network configuration to the source VM.
-          src_vm[:networks] = [
-            :networkName             => network_id,
-            :IsConnected             => true,
-            :IpAddressAllocationMode => "DHCP"
-          ]
+        }
+        if vm_opts.key?('disk_mb')
+          src_vm[:hardware][:disk] = vm_opts['disk_mb'].map { |disk| { :id => disk[:disk_id], :capacity_mb => disk[:value] } }
```

Comment:
Guessing you didn't one-line this because it was over 125 and rubocop complained :)

How about adding a `parse_disks` method so you can do:
`src_vm[:hardware][:disk] = parse_disks(vm_opts['disk_mb']) if vm_opts.key?('disk_mb')`